### PR TITLE
Fix bug bash comments

### PR DESCRIPTION
**Pull request by jalajmsft**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import * as path from 'path';
 
 import { createScriptFile, TEMP_DIRECTORY, NullOutstreamStringWritable, deleteFile, getCurrentTime } from './utils';
 
-const START_SCRIPT_EXECUTION_MARKER: string = 'Azure CLI GitHub Action: Starting script execution';
+const START_SCRIPT_EXECUTION_MARKER: string = `Starting script execution via docker image mcr.microsoft.com/azure-cli:`;
```
Comment (bishal-pdMSFT) [inline src/main.ts]: why do you want to print this statement inside container. Better keep it the way it was. and add the suffix part only before running container

File: src/main.ts [hunk 2]
```diff
@@ -47,18 +47,18 @@ const run = async () => {
         command += ` -v ${process.env.HOME}/.azure:/root/.azure -v ${TEMP_DIRECTORY}:${CONTAINER_TEMP_DIRECTORY} `;
         command += `-e GITHUB_WORKSPACE=${CONTAINER_WORKSPACE} --name ${CONTAINER_NAME}`;
         command += ` mcr.microsoft.com/azure-cli:${azcliversion} ${startCommand}`;
-        console.log(`${START_SCRIPT_EXECUTION_MARKER} via docker image mcr.microsoft.com/azure-cli:${azcliversion}`);
+        console.log(`${START_SCRIPT_EXECUTION_MARKER}${azcliversion}`);
```
Comment (bishal-pdMSFT) [inline src/main.ts]: As I said earlier, the previous code was better

File: src/main.ts [hunk 3]
```diff
@@ -47,18 +47,18 @@ const run = async () => {
...
     } catch (error) {
-        console.log("Azure CLI action failed.\n\n", error);
+        core.error(error);
         core.setFailed(error.stderr);
     }
     finally {
         // clean up
         const scriptFilePath: string = path.join(TEMP_DIRECTORY, scriptFileName);
         await deleteFile(scriptFilePath);
-        console.log("cleaning up conatiner");
```
Comment (bishal-pdMSFT) [inline src/main.ts]: put three dots after container

File: src/main.ts [hunk 4]
```diff
@@ -96,32 +96,37 @@ const executeDockerCommand = async (dockerCommand: string, continueOnError: bool
...
-            errline: (data: any) => {
-                if (data.toString().trim() === START_SCRIPT_EXECUTION_MARKER) {
-                    errorStream = ''; // Flush the container logs. After this, script error logs will be tracked.
+            errline: (data: string) => {
+                if (!shouldOutputErrorStream) {
+                    errorStream += data + os.EOL;
                 }
                 else {
-                    errorStream += data.toString() + os.EOL;
+                    console.log(data);
```
Comment (bishal-pdMSFT) [inline src/main.ts]: console.log or core.error? so that it shows up in red

File: src/main.ts (hunk 1 above)
Comment (jalajmsft) [reply src/main.ts]: echo inside the container is to draw a border between docker image download logs and user script logs. It is getting printed only once and not twice. If we remove it from within the container docker logs will also get printed

File: src/main.ts (hunk 4 above)
Comment (jalajmsft) [reply src/main.ts]: console.log was preferred as there can be warnings present in error stream, wherein the action still succeeds.

File: src/main.ts [hunk 5]
```diff
@@ -96,32 +96,37 @@ const executeDockerCommand = async (dockerCommand: string, continueOnError: bool
...
             }
         }
     };
-
+    var exitCode;
     try {
-        await exec.exec(`"${dockerTool}" ${dockerCommand}`, [], execOptions)
+        exitCode = await exec.exec(`"${dockerTool}" ${dockerCommand}`, [], execOptions)
     } catch (error) {
         if (!continueOnError) {
```
Comment (bishal-pdMSFT) [inline src/main.ts]: should you not check exitCode here as well?

Comment (bishal-pdMSFT) [reply src/main.ts]: sorry, it is fine

**Pull request merged**